src/Lecture4.hs

```diff
@@ -234,7 +293,37 @@ you can return string "no value"
 -}
 
 displayStats :: Stats -> String
-displayStats = error "TODO"
+displayStats stats =
+{-
+Total positions:       : 3
+Total final balance    : -15
+Biggest absolute cost  : 50
+Smallest absolute cost : 10
+Max earning            : 25
+Min earning            : 10
+Max spending           : 50
+Min spending           : 50
+Longest product name   : Pineapples
+-}
+    printf
+        "Total positions:       : %d\n\
+        \Total final balance    : %d\n\
+        \Biggest absolute cost  : %d\n\
+        \Smallest absolute cost : %d\n\
+        \Max earning            : %s\n\
+        \Min earning            : %s\n\
+        \Max spending           : %s\n\
+        \Min spending           : %s\n\
+        \Longest product name   : %s"
```

Comment:
Nice usage of `printf`! 👏🏻 
Usually, this function is discouraged because it's partial and doesn't check at compile-time that the number of arguments matches the required number of formatted strings. But works fine here 🙂